Backpack-SwiftUI/AppSearchModal/Classes/BPKAppSearchModal.swift

```diff
@@ -0,0 +1,152 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKAppSearchModal: View {
+    let title: String
+    @Binding var inputText: String
+    let inputHint: String
+    let results: BPKAppSearchModalResults
+    let closeAccessibilityLabel: String
+    let onClose: () -> Void
+    
+    public init(
+        title: String,
+        inputText: Binding<String>,
+        inputHint: String,
+        results: BPKAppSearchModalResults,
+        closeAccessibilityLabel: String,
+        onClose: @escaping () -> Void
+    ) {
+        self.title = title
+        self._inputText = inputText
+        self.inputHint = inputHint
+        self.results = results
+        self.closeAccessibilityLabel = closeAccessibilityLabel
+        self.onClose = onClose
+    }
+    
+    public var body: some View {
+        VStack(spacing: .base) {
+            
+            makeNavigationBar(title: title, closeAccessibilityLabel: closeAccessibilityLabel, onClose: onClose)
+            
+            switch results {
+            case .loading(let loading):
+                BPKTextField(placeholder: inputHint, $inputText)
+                AppSearchModalLoadingView(state: loading)
+            case .content(let content):
+                BPKTextField(placeholder: inputHint, $inputText)
+                AppSearchModalContentView(state: content)
+                    .padding(.top, .md)
+            case .error(let error):
+                AppSearchModalErrorView(state: error)
+                    .padding(.horizontal, .md)
+            }
+        }
+        .padding(.horizontal, .base)
+        .padding(.top, .base)
+        .padding(.bottom, 0)
+        .background(.surfaceElevatedColor)
+    }
+    
+    func makeNavigationBar(
+        title: String,
+        closeAccessibilityLabel: String,
+        onClose: @escaping () -> Void
+    ) -> some View {
+        ZStack {
+            HStack {
+                BPKIconView(.close, size: .large)
+                    .onTapGesture(perform: onClose)
+                    .accessibilityHidden(false)
+                    .accessibilityRemoveTraits(.isImage)
+                    .accessibilityAddTraits(.isButton)
+                    .accessibilityLabel(closeAccessibilityLabel)
+                Spacer()
+            }
+            BPKText(title, style: .heading5)
+                .accessibilityAddTraits(.isHeader)
+        }
+        .frame(height: 44)
```

Comment:
is there a way we could do this without a fixed height? perhaps with just paddings? so that if users use a bigger font size, this would not be a problem for them

Reply:
The vertical padding between the title label and the navigation bar is 12 in figma. 
So I added a padding of 4 to the title label and 8 to the navigation bar to comply with the backpack guideline.
Updated in [8fb8429](https://github.com/Skyscanner/backpack-ios/pull/1820/commits/8fb84290c463111b89ba819c3a2973704fc2d66c)